build-system/tasks/helpers.js

```diff
@@ -453,42 +457,46 @@ async function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
     };
   }
 
-  /**
-   * Bundles an entry point with all its imports and applies babel transforms.
-   * @param {boolean} continueOnError
-   */
-  async function performBundle(continueOnError) {
-    const {banner, footer} = splitWrapper();
-    await esbuild
-      .build({
-        entryPoints: [entryPoint],
-        bundle: true,
-        sourcemap: true,
-        define: experimentDefines,
-        outfile: destFile,
-        plugins: [esbuildBabelPlugin],
-        banner,
-        footer,
-      })
-      .catch((err) => handleBundleError(err, continueOnError, destFilename));
-    finishBundle(srcFilename, destDir, destFilename, options);
-    let name = destFilename;
-    if (options.latestName) {
-      const latestMaxName = options.latestName.split('.js')[0] + '.max.js';
-      name = `${name} → ${latestMaxName}`;
-    }
-    endBuildStep('Compiled', name, startTime);
-    const target = path.basename(destFilename, path.extname(destFilename));
-    if (UNMINIFIED_TARGETS.includes(target)) {
-      await applyAmpConfig(
-        destFile,
-        /* localDev */ true,
-        /* fortesting */ options.fortesting
+  const {banner, footer} = splitWrapper();
+  async function onRebuild(error, result) {
+    if (error) {
+      log.error(
+        red(`Error while recompiling ${destFilename}`),
+        `: ${error.message}`
       );
+      return;
     }
+
+    log(green(`Recompiled ${destFilename}.`));
+    if (result.warnings.length > 0) {
+      for (const warning of result.warnings) {
+        log.warn(`${yellow('Warning')} in ${warning.file}: ${warning.text}`);
+      }
+    }
+    await postBundle(srcFilename, destDir, destFilename, options);
   }
 
-  await performBundle(options.continueOnError);
+  await esbuild
+    .build({
+      entryPoints: [entryPoint],
+      bundle: true,
+      sourcemap: true,
+      define: experimentDefines,
+      outfile: destFile,
+      plugins: [esbuildBabelPlugin],
+      banner,
+      footer,
+      incremental: options.watch,
+      watch: options.watch ? {onRebuild} : false,
+      logLevel: 'silent',
+    })
+    .catch((err) =>
+      handleBundleError(err, options.continueOnError, destFilename)
+    );
+  const name = options?.latestName?.replace(/\.js$/, '.max.js') ?? destFilename;
```

Comment:
This logic seems broken. I'm seeing `Compiled amp.js → amp.js`, which isn't supposed to happen because `amp.js` is not an extension and doesn't have a `lastestName` property.

Reply:
I probably misunderstood what that log was meant for! The intention was to print the `Compiling X --> X.js` for _just_ extensions? I'll bring back the latestName check

Reply:
I actually like this behavior more than current. It consistently will display `fromFile -> toFile` instead of the mixed results in master where I sometimes see a line like: `amp-loader-0.1.max.js → amp-loader-latest.max.js`. 

Maybe I'm misunderstanding what the arrow means?

Reply:
The arrow indicates which version of an extension maps to the "latest" version. For example:

![image](https://user-images.githubusercontent.com/26553114/109592057-a552e880-7adc-11eb-949e-2a41cb425d74.png)

Reply:
Ah! Ok I did not realize the arrow was meant to indicate the `copySync` that happens with latest renames. I'll update